core/Container/ContainerFactory.php

```diff
@@ -82,4 +82,16 @@ private function addEnvironmentConfig(ContainerBuilder $builder)
 
         $builder->addDefinitions($file);
     }
+
+    private function addPluginConfigs(ContainerBuilder $builder)
+    {
+        $plugins = Config::getInstance()->Plugins['Plugins'];
```

Comment:
Not sure, it *should* be safe. Needs to be tested and checked what is done in there, only had a rough look. 

Also we need to make sure that during this process DI is not used in the background eg by Config etc? Otherwise it might end in a problem again... 

Maybe it makes sense to load "global.php" first... while loading/activating the plugins we also load their definitions. Might be easiest solution that *should* work and also makes sure we load it only if needed eg while tests are run etc. On "deactivate Plugin" we could maybe unload the definition - if possible. Not really needed though.